File: Houston/src/main/java/pl/tajchert/houston/NotificationWrapper.java
package pl.tajchert.houston;


public class NotificationWrapper {
  public int id;
  public String tag;
  public NotificationCategory category;
  public Long showTime;

  public boolean equals(String tag) {
    return tag != null ? tag.equals(this.tag) : this.tag == null;
  }

  public boolean equals(NotificationCategory notificationCategory) {
    if (this.category == null) return false;
    return notificationCategory != null && category.equals(notificationCategory);
  }

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    NotificationWrapper yawn = (NotificationWrapper) o;
    return id == (yawn.id);
  }
}

File: library/src/main/java/pl/tajchert/houston/NotificationWrapper.kt
package pl.tajchert.houston

class NotificationWrapper {
  var id: Int = 0
  var tag: String? = null
  var category: NotificationCategory? = null
  var showTime: Long? = null

  fun equals(tag: String?): Boolean {
    return if (tag != null) tag == this.tag else this.tag == null
  }

  fun equals(notificationCategory: NotificationCategory?): Boolean {
    return if (this.category == null) false else notificationCategory != null && category == notificationCategory
  }

  override fun equals(o: Any?): Boolean {
    if (this === o) return true
    if (o == null || javaClass != o.javaClass) return false
    val yawn = o as NotificationWrapper?
    return id == yawn!!.id
  }
}
